neuraxle/steps/sklearn.py

```diff
@@ -78,9 +78,11 @@ def transform(self, data_inputs):
             return self.wrapped_sklearn_predictor.predict(data_inputs)
         return self.wrapped_sklearn_predictor.transform(data_inputs)
 
-    def set_hyperparams(self, flat_hyperparams: HyperparameterSamples) -> BaseStep:
-        BaseStep.set_hyperparams(self, flat_hyperparams)
-        self.wrapped_sklearn_predictor.set_params(**HyperparameterSamples(flat_hyperparams).to_flat_as_dict_primitive())
+    def _set_hyperparams(self, hyperparams: HyperparameterSamples, root=False) -> BaseStep:
+        BaseStep._set_hyperparams(self, hyperparams)
+        step_hyperparams, remainders = BaseStep._create_recursive_step_values_and_remainders(self, hyperparams, root)
+        self.wrapped_sklearn_predictor.set_params(**HyperparameterSamples(step_hyperparams).to_flat_as_dict_primitive())
+        self.wrapped_sklearn_predictor.set_params(**HyperparameterSamples(HyperparameterSamples(remainders).to_flat()).to_flat_as_dict_primitive())
```

Comment:
It would be good here to split these lines on multiple lines for easier reading of the code.

Reply:
Also, questions here... 
1. Why do you call 2 times the set_params(...) here?
2. Why do you call 3 times the HyperparameterSamples(...) here ?

Reply:
1. because of the way it was done before I had to do that. 
2. to flatten them

Reply:
this assert would not pass if I wouldn't call set_params for the remainder : 
```python
    import sklearn.pipeline
    the_step = SomeStep()
    step_to_check = the_step.tosklearn()

    p = Pipeline([
        ("a", SomeStep()),
        ("b", SKLearnWrapper(sklearn.pipeline.Pipeline([
            ("a", sklearn.pipeline.Pipeline([
                ('z', step_to_check)
            ])),
            ("b", SomeStep().tosklearn()),
            ("c", SomeStep().tosklearn())
        ]), return_all_sklearn_default_params_on_get=True)),
        ("c", SomeStep())
    ])

    # assert False
    p.set_hyperparams({
        "b": {
            "a__z__learning_rate": 7,
            "b__learning_rate": 9
        }
    })
    assert p.get_hyperparams()['b__a__z__learning_rate'] == 7

```